manageiq-setup.sh

```diff
@@ -21,31 +21,55 @@ EOF
 /usr/sbin/semanage fcontext -a -t httpd_log_t "/var/www/miq/vmdb/log(/.*)?"
 /usr/sbin/semanage fcontext -a -t cert_t "/var/www/miq/vmdb/certs(/.*)?"
 /usr/sbin/semanage fcontext -a -t logrotate_exec_t ${APPLIANCE_SOURCE_DIRECTORY}/logrotate_free_space_check.sh
-/usr/sbin/semanage fcontext -a -t etc_t "/var/www/miq/vmdb/config/cockpit"
-
-cat <<'EOF' > /tmp/cockpit_ws_port.te
-module cockpit_ws_port 1.0;
+/usr/sbin/semanage fcontext -a -t cockpit_ws_exec_t "/var/www/miq/vmdb/tools/cockpit/cockpit-auth-miq"
 
+cat <<'EOF' > /tmp/cockpit_ws_miq.te
+module cockpit_ws_miq 1.0;
 require {
     attribute port_type;
-	type cockpit_ws_t;
-	class tcp_socket { name_bind };
+    type cockpit_ws_t;
+    type bin_t;
+    type usr_t;
+    type user_home_dir_t;
+    type ephemeral_port_t;
+    type httpd_sys_content_t;
+    type proc_t;
+    class capability dac_read_search;
+    class tcp_socket name_connect;
+    class lnk_file { getattr read };
+    class file { execute execute_no_trans getattr ioctl open read };
+    class dir { read write };
+    class tcp_socket { name_bind };
+    class dir audit_access;
 }
 
-type cockpit_ws_port_t, port_type;
-allow cockpit_ws_t cockpit_ws_port_t:tcp_socket name_bind;
+type cockpit_ws_miq_t, port_type;
+allow cockpit_ws_t cockpit_ws_miq_t:tcp_socket name_bind;
+
+allow cockpit_ws_t bin_t:file execute;
+allow cockpit_ws_t usr_t:file { execute execute_no_trans };
+
+allow cockpit_ws_t ephemeral_port_t:tcp_socket name_connect;
+allow cockpit_ws_t httpd_sys_content_t:dir read;
+allow cockpit_ws_t httpd_sys_content_t:file { getattr ioctl open read };
+allow cockpit_ws_t httpd_sys_content_t:lnk_file { getattr read };
+allow cockpit_ws_t proc_t:file { getattr open read };
+allow cockpit_ws_t self:capability dac_read_search;
+dontaudit cockpit_ws_t user_home_dir_t:dir audit_access;
+dontaudit cockpit_ws_t usr_t:dir audit_access;
 EOF
-checkmodule -M -m -o /tmp/cockpit_ws_port.mod /tmp/cockpit_ws_port.te
-semodule_package -o /tmp/cockpit_ws_port.pp -m /tmp/cockpit_ws_port.mod
-semodule -i /tmp/cockpit_ws_port.pp
 
-semanage port -a -t cockpit_ws_port_t -p tcp 9002
+checkmodule -M -m -o /tmp/cockpit_ws_miq.mod /tmp/cockpit_ws_miq.te
+semodule_package -o /tmp/cockpit_ws_miq.pp -m /tmp/cockpit_ws_miq.mod
+semodule -i /tmp/cockpit_ws_miq.pp
+
+semanage port -a -t cockpit_ws_miq_t -p tcp 9002
```

Comment:
I'm so lost.  Was this broken before?  Is this PR fixing a bug, regression, or adding features?  I honestly don't know.  From the size of this PR, it seems like parts of this were never working.  

The commit message doesn't help me understand why there are 50 net lines changed here.

I'm concerned by the use of ports and sockets here because we'll have to rewrite most if not all of this communication code to use services in openshift in the rearchitecture.

Reply:
@jrafanie this is updating the selinux rules to work with the updated OS and selinux rules. The previous version worked with a previous version of the appliance but does not work with the latest images. There is no new functionality here.

The part that opens port 9002 is not new. https://github.com/ManageIQ/manageiq-appliance/blob/master/manageiq-setup.sh#L42 Of course this will no longer be needed when you rearchitecture as ports and routing are handled by openshift and there is no need to tell selinux about them.

Reply:
> @jrafanie this is updating the selinux rules to work with the updated OS and selinux rules. The previous version worked with a previous version of the appliance but does not work with the latest images. There is no new functionality here.

Ok, this makes sense.  I wasn't aware we made a big change in OS.  Do we know what version worked previously?  Did we update selinux or another related package that broke us?  I'm not sure how you do integration tests of this.  I guess, if this were to break in the future for a user, how would we triage what package versions are important or anything that would help us determine what might have broken us?

Reply:
I believe I last tested with 7.1. So that was a while ago. I'm not sure how these appliances are tested. Are there any automated tests run anywhere against the generated artifacts? How do you currently test for selinux issues? If there are any e2e tests like that, I'm happy to try to to integrate with that before getting this in.

Reply:
Appliances are tested by QE but we have nothing in travis to test this.  We don't often change selinux related code but try to explain why it's changed for tracking and historical purposes.  I'm fine with making changes here but ultimately, we'll need to understand how a future change in manageiq or selinux could break any of these.

For example, I don't understand the changes here well enough to know if and how we'd implement them when manageiq is run as containers.  If someone else is comfortable with these changes, I'm fine with merging this.

Reply:
The Selinux policy packages change frequently and cause breakage across all sorts of apps. For example the havoc that the new 'map' permission is causing in fedora. Or even just the semi-regular regressions we see running tests on selinux machines.

So beside selinux changing things, from the app side changes that could cause this need to change again here could include.
 * MIQ content no longer marked as httpd_sys_content_t
 * Ruby changing what it does when launching a new process
 * The miq logging setup changing.